ModMenu/NewTypes/SettingsEntityPatches.cs

```diff
@@ -329,6 +330,17 @@ private static SettingsEntitySubHeaderView CreateSubHeaderTemplate(GameObject pr
 
         return templatePrefab;
       }
+
+      [HarmonyPatch(typeof(SettingsVM))]
+      static class ApplySettings_Patch
+      {
+        [HarmonyPatch(nameof(SettingsVM.ApplySettings)), HarmonyPostfix]
+        static void Postfix()
+        {
+          // Raise Event when Apply Settings button is press in the settings UI
+          EventBus.RaiseEvent<ISettingsChanged>(h => h.HandleApplySettings());
```

Comment:
Any reason not to pass the setting that was changed here?

Reply:
I was using it just detect IF the settings ANY were applied. I needed to rebuild some tables if anything was changed and didn't want to have OnValueChanged on all my settings.